GoogleForCreators/web-stories-wp
# Render "sizes" attribute for amp-img

## Context

`sizes` should always accompany `srcset` attribute for responsive image selection. We were relying on AMP's autogeneration of `sizes` which sometimes yields bad results (#7500).

## Summary

Render `sizes` attribute for image output (amp-img) for desktop and mobile viewing modes.

## Relevant Technical Choices

Use the image element width to determine its size in the story.

## To-do

The [`ApplyCommonAttributes`](https://github.com/ampproject/amp-toolbox/blob/main/packages/optimizer/lib/transformers/ApplyCommonAttributes.js) AMP optimizer transform (part of the `ServerSideRendering` transform) moves `sizes` to CSS media queries, which means the `amp-img[sizes]` autogeneration still happens. I assume this is WAI but needs more testing.

## User-facing changes

`srcset` selection should be more consistent and never result in blurry (low resolution) images.

## Testing Instructions



### QA



This PR can be tested by following these steps:

1. Insert images with `srcset` into a story (e.g. Unfold images).
2. Preview the story across a variety of screen widths.
3. Make sure images look reasonable (not blurry).

### UAT






## Reviews

### Does this PR have a security-related impact?

No.

### Does this PR change what data or activity we track or use?

No.

### Does this PR have a legal-related impact?

No.

## Checklist




---



Fixes #7500
